test/grpc/integration/server_test.exs

```diff
@@ -7,6 +7,12 @@ defmodule GRPC.Integration.ServerTest do
     def get_feature(point, _stream) do
       %Routeguide.Feature{location: point, name: "#{point.latitude},#{point.longitude}"}
     end
+
+    def route_chat(_ex_stream, stream) do
+      GRPC.Server.send_headers(stream, %{})
```

Comment:
Hi, I'm investigating a Cowboy shutdown issue for my gRPC service running version v0.9.0 in production. My client receives `{:error, %GRPC.RPCError{status: 13, message: "status got is 204 instead of 200"}}` when Cowboy shuts down.

In my unit test, if I add this line, my client receives `{:error, %GRPC.RPCError{status: 13, message: ":stream_error: {:stream_error, :no_error, :\"Stream reset by server.\"}"}}`.

How can I gracefully handle the service's endpoint shutdown from the client side? Should I send headers like retry or goaway to prompt the client to retry?